[Team-21, 노을] baseball 기능 구현 및 배포

sanhee opened the pull request:
**프론트 배포 주소**: http://baseball.san.r-e.kr/
**API 문서**:  https://documenter.getpostman.com/view/14857650/TzRSho8L
**DB 구조**:  https://github.com/sanhee/baseball/wiki/DB-%EB%AA%A8%EB%8D%B8%EB%A7%81

### 한 일

- [X] 팀원과 협의한 API 구현
- [X] public, private ec2 인스턴스 배포
- [X] 커스텀 도메인에 public ec2  IP 연동
- [x] 배포 자동화 스크립트 (crontab 이용)
- [x] Nginx 를 이용한 배포
  -  80포트로 프론트 / API 모두 접근 가능
- [ ] 예외처리, 클린코드, 테스트코드, OAuth

----

안녕하세요, 백엔드반 노을입니다.

이번 야구게임 프로젝트를 혼자서 하다보니 갈피를 잡기가 어려워, 
돌아가는 코드와 클라우드 배포 경험을 중점적으로 프로젝트를 진행하였습니다.
( 코드상태가 매우 좋지 못합니다 ㅠㅠㅠ.. )

초기 객체 설계를 할 때 Game 레포지토리에서 모든 하위 엔티티를 관리할 수 있게끔 설계를 해보았는데,
상위부터 하위까지의 참조 로직이 너무 복잡해져서 개발시간이 지체돼 커스텀 쿼리 메소드를 난발하게 됐습니다.

질문 드릴점은 아래와 같이 Game Repostiory에서 반환값으로 Game외에 다른 객체들과 DTO들을 전달하고 있는데 괜찮은 방식인지 궁금합니다.

또 컨트롤러의 response 반환값으로  주로  `List<DTO>` 형태를 사용하고 있는데 이 부분을 Wrapping 하라는 의미를 잘모르겠어서 약간의 도움을 주셨으면 좋겠습니다. 😁

이상으로 오늘도 바쁜 시간내어 리뷰 해주셔서 감사합니다!


```java
public interface GameRepository extends CrudRepository<Game, Long> {

    Optional<Game> findById(Long id);

    List<Game> findAll();

    @Query ...
    Optional<Game> findGameById(@Param("id") Long id);

    @Query ...
    Optional<Team> findTeamByTitle(@Param("title") String title);

    @Query ...
    Optional<Player> findPlayerByName(@Param("name") String name);

    @Query ...
    List<GameList> findAllGame();
```

sanhee committed:
Update README.md

sanhee committed:
feat: 스프링 부트 프로젝트 생성

sanhee committed:
[#6] feat: ER 다이어그램 작성

sanhee committed:
chore: .gitignore 수정

sanhee committed:
[#10] feat: DB 스키마 및 더미 데이터 SQL 작성

sanhee committed:
[#10] feat: Game 객체 구현

sanhee committed:
[#10] Team Domain 생성 및 1:N 관계설정

Game이 Team을 가지고 있는 1:N관계 입니다.
팀 리스트를 한꺼번에 보여줘야 하므로 List 로 관계설정을 하였습니다.
1:N관계를 설정할떄 key_column을 필드 추가없이 Team의 아이디로 지정하였습니다.

sanhee committed:
[#10] feat: Player Domain 생성 및 1:N 관계 설정

Team이 Player를 복수개 가지고 있는 것으로 설계하여 1:N 설정을 하였습니다.

sanhee committed:
[#10] feat: PlayerHistory Doamin 생성 및 1:N 관계 설정

Player의 History는 이닝마다 여러 데이터가 생길 수 있으므로, 1:N으로 관계를 맵핑하였습니다.

sanhee committed:
[#10] refactor: team 검색용 쿼리 메서드 정의

자바로 타고타고 들어가다보니 너무 복잡해져서 for문을 줄이고나 쿼리 메서드를 이용했습니다.

sanhee committed:
[#10] feat: TeamScore Domain 생성 및 1:N 관계 설정

이닝마다 Team의 점수를 저장해야하기 때문에, 1:N으로 맵핑하였습니다.

sanhee committed:
[#10] chore: PlayerScore 클래스가 삭제됨에 따라 더미 데이터 수정

PlayerScore가 1:1 관계인걸 까먹고 1:N으로 맵핑을 하다가 삽질을 너무많이 했네요 ㅎㅎ
우선 Player에 합치고, 나중에 분리할 생각입니다.

sanhee committed:
[#10] chore: 코드 정리

sanhee committed:
Merge pull request #15 from sanhee/feature/implement-entity

DB객체 맵핑

sanhee committed:
[#14] feat: 게임 리스트 API 구현

각 게임번호(ID)에 해당하는 원정, 홈팀을 출력하는 API입니다.

sanhee committed:
[#14] fix: Game 리스트 API에 inProgress 필드 추가

- 게임리스트를 제공할 때, 팀 선택에 대한 정보는 필요없으므로, 제거

sanhee committed:
[#14] feat: 팀 선택 API 구현

- 팀 선택을 하게될 경우, 해당 팀의 selected 필드가 true로 변경됩니다.
- 선택한 팀과 동일한 게임아이디를 참조하고 있는 팀을 조회합니다.

sanhee committed:
[#14] feat: 게임중 Player Score, Team Score 저장하는 API 구현

자바 코드가 복잡해지는 것을 막기위해, 쿼리 메서드를 활용

sanhee committed:
[#14] chore: Dto 패키지 경로 정리

sanhee committed:
[#14] feat: 상세 점수 팝업 API 구현

sanhee committed:
[#14] fix: 팀 선택 API 기존 요구사항 형식에 맞게 개선

sanhee committed:
[#14] fix: 팀 선택 API 사용시 게임 상태가 변경되지 않는 점 개선

팀 selected와 게임의 상태를 동시 업데이트

sanhee committed:
[#14] feat: 선수 명단 팝업 API 구현

sanhee committed:
[#14] fix: 선수 업데이트 요청 정보 JSON 형식으로 변경

sanhee committed:
Merge pull request #32 from sanhee/implement-api

API 구현

sanhee committed:
[#36] feat: 타자 정보 업데이트 요청 반환 값으로 다음 플레이어 정보 전달

- 기능 구현을 목적으로 클린코드, 예외처리 모두 배제하고 코드를 작성하였습니다.

sanhee committed:
[#36] refactor: controller 내부로직 service 로 이동

sanhee committed:
Merge pull request #40 from sanhee/feature/nextPlayer-implement

타자 정보 업데이트 요청 후 리스폰스 구현

sanhee committed:
chore: spring.datasource.url DB서버로 수정

sanhee committed:
chore: UPDATE README.md

sanhee committed:
chore: UPDATE data.sql

일부 기본값 정보가 잘못돼서, 수정했습니다.

sanhee committed:
chore: CORS 옵션 추가

sanhee committed:
chore: CORS 옵션 삭제

sanhee committed:
chore: UPDATE data.sql

sanhee committed:
chore: CORS 설정

sanhee committed:
chore: CORS 설정

sanhee committed:
feat: reset API 기능 추가

sanhee committed:
chore: UPDATE README.md

remote 버전과 공백차이가 나서 수정했습니다.

sanhee committed:
refactor: GameRepository 메소드 관련있는 것끼리 정렬

sanhee committed:
refactor: DTO 이름 제거

패키지명에 이미 DTO가 있어서 제거했습니다.

sanhee committed:
feat: 배포 자동화 스크립트 추가

- git의 버전을 비교하면서, 차이가 생길 경우 머지 및 빌드를 해주는 스크립트

sanhee committed:
chore: application.properties 불필요한 정보 삭제

sanhee committed:
feat: 브랜치 병합

sanhee committed:
refactor: 자주쓰이는 상수 final 정의

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+import java.util.Objects;
+
+@RestController
+@RequestMapping("/api/")
+@CrossOrigin(origins = "*", allowedHeaders = "*")
```
모든 API 요청이 이 컨트롤러를 거친다면, 한 컨트롤러가 너무 많은 책임을 지진 않을까요?

ksundong replied:
CrossOrigin 이슈를 해결하는 다른 방법도 있는데 한 번 알아보셨으면 좋겠습니다.
CORS에 대해서는 면접 질문으로도 나오니만큼 잘 알아보세요.

sanhee replied:
CORS에 대해서 학습한걸 요약해봤는데, 혹시 잘 정리한게 맞을까요??

`CORS`는 서버의 문제가 아니다. `CORS`는 브라우저가  `CSRF, XSS` 등 신뢰할 수 없는 요청에서 사용자를 보호하기 위해 막는 것이다.

서버의 응답 헤더에 `Access-Control-Allow-Origin` 설정이 있고, 
프론트가 API를 요청할 때 `출처(origin)`를 서버가 허용하는 출처와 맞춰주지 않으면 `CORS` 에러가 발생할 수 있다.

프론트 개발자가 로컬에서 개발할 때 대안으로 프록시로 origin을 위장해서 API 통신할 수 있음

sanhee replied:
> 모든 API 요청이 이 컨트롤러를 거친다면, 한 컨트롤러가 너무 많은 책임을 지진 않을까요?

저한테 아직도 어색하고 어려운 개념이라 잘못 적용한 부분 같습니다.

DDD 모델링의 연관 객체의 묶음을 Aggregate Root에서 관리한다고 생각해서, 이거랑 비슷한 맥락으로 Aggregate Root 컨트롤러에서 모든 걸 제어하도록 코드를 작성을 했는데 잘못된 방식인건가요??

어떻게 적용해야할지.. 감이 잘 안오고 어려운 것 같습니다..😂😂

ksundong replied:
CORS는 서버의 문제가 될 수도 있습니다. Host를 허용을 해줘야 하는데 하지 않아서 발생할 수도 있으니까요.
좀 더 깊이 공부하시는걸 추천드려요. 실무에서도 많이들 겪는 문제입니다.
미리 알아두면 고생을 덜하겠죠? ㅎㅎ

ksundong replied:
Aggregate 개념이 도메인에만 적용된다고 생각하시면 좋을 것 같아요.
모든걸 제어한다는게 좋은것이 아닙니다.
Aggregate의 경우에는, 전체의 조작을 해당 aggregate의 주인인 root에서 함으로써 사이드 이펙트를 방지한다는 장점이 있지만
컨트롤러에서는 굳이 그러지 않아도 된다고 생각해요.

sanhee replied:
> CORS는 서버의 문제가 될 수도 있습니다. Host를 허용을 해줘야 하는데 하지 않아서 발생할 수도 있으니까요.
> 좀 더 깊이 공부하시는걸 추천드려요. 실무에서도 많이들 겪는 문제입니다.
> 미리 알아두면 고생을 덜하겠죠? ㅎㅎ

앗.. 너무 가볍게 공부한게 바로 들통 나버렸네요.. 죄송합니다...
해당 내용은 좀더 공부해서 정리해보는 시간을 가져보도록 하겠습니다.

sanhee replied:
> Aggregate 개념이 도메인에만 적용된다고 생각하시면 좋을 것 같아요.
> 모든걸 제어한다는게 좋은것이 아닙니다.
> Aggregate의 경우에는, 전체의 조작을 해당 aggregate의 주인인 root에서 함으로써 사이드 이펙트를 방지한다는 장점이 있지만
> 컨트롤러에서는 굳이 그러지 않아도 된다고 생각해요.

해당 부분도 좀 더 많은 공부가 필요한 것 같네요.. 답변 해주셔서 감사합니다!
컨트롤러를 역할에 따라 분리해보도록 하겠습니다.

sanhee replied:
혹시 
https://github.com/codesquad-members-2021/baseball/pull/44/commits/35e016331085b977a3810307fd7b5b35e7ed2335

이렇게 범용적인 Wrapping 객체를 사용하면 되는걸까요?!

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+
+    private final GameService gameService;
+
+    @Autowired
```
이제 이 애노테이션이 없어도 된다는 것은 알고계시겠죠?

sanhee replied:
생성자 주입을 할 때, 단일 생성자에서는 생략이 가능한 것으로 알고있습니다!

하지만, 코드를 작성할 때는 학습 목적으로 잊지 않기위해서 명시적으로 붙여줬습니다!!

ksundong replied:
잊으셔도 됩니다. 필요할 때 찾아서 쓰면 됩니다.

sanhee replied:
아하 알겠습니다! 바로 지웠습니다 ㅎㅎ

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+        this.gameService = gameService;
+    }
+
+    @GetMapping("game_list")
```
보통 url에선 언더스코어를 거의 쓰지 않습니다.
이는 예전 관습 때문인데요. 링크에는 밑줄이 표시되기 때문에 그것이 구분되지 않는다 하여 하이픈을 많이 씁니다.

ksundong replied:
그리고 저라면 games로 표현해줄 것 같아요.

sanhee replied:
오호 처음 알게 됐습니다! URL 이름을 명명할 때 스네이크케이스와 카멜케이스를 고민했었는데 그런 관습이 있군요!! 감사합니다!

https://github.com/codesquad-members-2021/baseball/pull/44/commits/8b4af4ce0ba018e7bd0bf59f2dee27877d7eb1d4

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    @GetMapping("game_list")
+    private List<GameList> gameList() {
```
List형태로 리턴하는 API는 변경에 유연하지 않습니다.
다른 분들 리뷰에 꽤 많이 달았었는데, 서로간에 리뷰 받은 내용이 공유된다면 더 좋을 것 같아요.

ksundong replied:
또한 GameList의 List인가요?
어떤 이유 때문일지 궁금하네요.

sanhee replied:
> 또한 GameList의 List인가요?
> 어떤 이유 때문일지 궁금하네요.

GameList가 각 gameId마다의 팀을 조회하는데, 이름을 잘못 명명한 것 같습니다..
GameList -> GameInfo 로 수정했습니다.

```json
[
    {
        "game_id": 1,
        "in_Progress": true,
        "home": "Backend1",
        "away": "Backend2"
    },
    {
        "game_id": 2,
        "in_Progress": true,
        "home": "Frontend1",
        "away": "Frontend2"
    },
    {
        "game_id": 3,
        "in_Progress": false,
        "home": "BE+FE",
        "away": "IOS1"
    }
]
```

sanhee replied:
> List형태로 리턴하는 API는 변경에 유연하지 않습니다.
> 다른 분들 리뷰에 꽤 많이 달았었는데, 서로간에 리뷰 받은 내용이 공유된다면 더 좋을 것 같아요.

Response 클래스를 만들어서, 객체를 묶어서 리턴했습니다!
```java
Response<List<GameInfo>> gameList();
```

이러한 형식으로 했는데, 올바르게 했는지 궁금합니다!

https://github.com/codesquad-members-2021/baseball/pull/44/commits/2161475178962b43f44819de2b6e5fc0b4b0a77f

ksundong replied:
당연히 좋지 않습니다. ㅎㅎ
일관된 응답은 예외정도만 일관적으로 해주시고, 적절한 Wrapper 객체를 만들어주세요.

ksundong replied:
물론 협의에 따라 다르겠지만, 저는 일반적으로 저런 형태는 2중으로 작업을 해야하기 때문에 별로라고 생각해요.

sanhee replied:
리스폰스를 범용적으로 리턴할 수 있는 Wrapper 객체를 만들라고 말씀해주신건가요?!!
제가 이해력이 좋지 못해서 적절한 Wrapper 객체가 무엇인지 잘 모르겠어서....어떻게 해야할지 잘 모르겠습니다 ㅠㅠ.....
조금만 더 힌트를 주시면 이해하는데 큰 도움이 될 것 같습니다...!

ksundong replied:
DM으로도 말씀드렸지만, 범용 Response는 그다지 좋은 형태가 아닌것 같다는 의견을 드립니다~

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+    @GetMapping("game_list")
+    private List<GameList> gameList() {
+        logger.debug("gameService.findAllGame() : {}", gameService.findAllGame());
+        return gameService.findAllGame();
```
호출을 두 번 하는 구조는 별로 바람직하지 못한 것 같네요.

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/827cde6112e0de847ddfd99c9982ef609155dae0

개선했습니다!

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+        return gameService.findAllGame();
+    }
+
+    @GetMapping("select_team/{title}")
```
이 Url의 구조는 다시 생각해보시기 바랍니다.

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/a8d1192151abcf12ef58b1926f36f95c8416fcba

업데이트를 시키는 로직으로, `GET` 보다는 `PUT`이 어울린다고 생각하여
변경하였습니다!!

ksundong replied:
url도 한 번 생각해보세요!

sanhee replied:
> url도 한 번 생각해보세요!

PUT을 사용할 떄 불필요한 동사를 자제하라는 리뷰를 떠올리며, `@PutMapping("team")` 와 같이 수정을 하였습니다!

ksundong replied:
그러면 team을 수정하는 형태가 될텐데요.
저라면 POST를 쓰고, team/select 정도로 시도해봤을 것 같아요.

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    @GetMapping("select_team/{title}")
+    @ResponseStatus(HttpStatus.OK)
```
기본이 200응답입니다.

sanhee replied:
앗.. 뭔가 명시적으로 해줘야 할 것 같은 느낌이었는데 아니었나 보군요!!
삭제 완료했습니다!

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+    @GetMapping("select_team/{title}")
+    @ResponseStatus(HttpStatus.OK)
+    private List<TeamListDto> selectGame(@PathVariable String title) throws Exception {
+        title = Objects.toString(title, "");
```
이렇게 해주신 이유는 무엇인가요?

sanhee replied:
혹시나 null 값이 들어왔을때 처리하려고 무의식적으로 빈 문자열을 지정해줬었는데, 빈 문자열이 아닌 에러를 발생시키는게 옳은 방식인 것 같습니다.

ksundong replied:
null일 경우가 있을까요?

sanhee replied:
> null일 경우가 있을까요?

생각해보니 url에서 null을 받을 수 있는 경우는 존재하지 않을 것 같습니다.
생각없이 작성하다보니 이런 불상사가 일어났네요.. 
지적해주셔서 감사합니다.

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+        return gameService.getInfoSelectedTeam(title);
+    }
+
+    @PutMapping("/update_player")
```
이미 PUT이 Update의 의미를 포함하고 있습니다.
가급적이면 url에는 불필요한 동사가 등장하지 않게끔 유지해주세요.

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/fe1ad4d74b53c87a90668f072293506e86d913dd

크게 생각하지 못했던 부분인데 지적해주셔서 감사합니다!!

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    @PutMapping("/update_player")
+    @ResponseStatus(HttpStatus.CREATED)
```
좋은 사용이에요.

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+
+    @PutMapping("/update_player")
+    @ResponseStatus(HttpStatus.CREATED)
+    private NextPlayerInfoDto updatePlayerInfo(@RequestBody UpdatePlayerInfo req) throws Exception {
```
약어 사용은 지양해주시고, 왜 Exception을 throw하는지 잘 이해가 되지 않네요.

sanhee replied:
안좋은 버릇이 이곳저곳에 있네요.. 의식적으로 훈련을 해야겠습니다..

Exception의 경우도 확인해보니 불필요한 상황이어서 제거하였습니다.

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+        return gameService.updatePlayerInfo(req);
+    }
+
+    @GetMapping("/detailScore/{gameID}")
```
어디는 언더바를 사용하고 어디는 카멜케이스를 사용하는건 일관적이지 않습니다.

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/8b4af4ce0ba018e7bd0bf59f2dee27877d7eb1d4
하이폰으로 통일 시켰습니다. 지적 감사합니다!

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+
+    @GetMapping("/playerList/{gameId}")
+    @ResponseStatus(HttpStatus.OK)
+    private List<ScoreList> playerScoreList(@PathVariable Long gameId) throws Exception {
```
접근제어자가 왜 다 private인지 궁금합니다.

sanhee replied:
외부에서 사용되는 곳이 없고, 접근제어자를 정하기 애매할 때는 `private` 으로 우선 정하고 필요할 때 오픈하는게 좋다고 들은것 같아서 작성을 했었습니다. 혹시 문제가 있다면 알려주시면 감사하겠습니다!

ksundong replied:
Controller는 EndPoint이기 때문에, 외부에 열려있어야 하지 않을까요?

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+    @ResponseStatus(HttpStatus.OK)
+    private List<ScoreList> playerScoreList(@PathVariable Long gameId) throws Exception {
+        Game findGame = gameService.findGameById(gameId);
+        return gameService.getPlayerScoreOfGame(findGame);
```
참신한 접근이긴한데, 그냥 `gameService.getPlayerScoreOfGame`에 gameId를 넘겨주면 되지 않을까요?

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/43b9361a4e9251765da31dd4b4ad33b01cd0ff91

컨트롤러에 있던 게임 탐색 로직을 서비스로 옮기라는 말씀으로 이해했는데 혹시 잘이해한게 맞을까요?..

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+
+    @GetMapping("/detailScore/{gameID}")
+    @ResponseStatus(HttpStatus.OK)
+    private List<DetailScore> detailScore(@PathVariable Long gameID) {
```
어디는 gameID고, 어디는 gameId네요. 일관성 있는 코드를 작성해주세요.

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/60c5564e60783e7626f4167588ea2d7ba9424770

수정완료했습니다!!
의식적으로 카멜케이스를 지키도록 노력하겠습니다. 😀

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+        return gameService.getPlayerScoreOfGame(findGame);
+    }
+
+    @PutMapping("/reset")
```
보통 이런 경우에는 POST를 써주시는게 좋습니다.
명확하지 않은 경우엔 POST를 사용한다라고 생각하시면 좋을 것 같아요.

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/3c5e369f9659e5635360965a93ce2447ae442734

POST의 경우 새로운 리소스 생성, PUT 보다는 좀 큰 범위의 수정으로 생각하면 될까요?!

ksundong replied:
POST는 새로운 리소스 생성도 있지만, REST가 커버 불가능한 모든 범위의 행동을 수행할 때 사용해주시면 됩니다.

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    @PutMapping("/reset")
+    @ResponseStatus(HttpStatus.RESET_CONTENT)
```
해당 HTTP Status를 사용한 이유는 알겠습니다만, void를 리턴한다면 204를 추천드리고
200응답도 나쁘지 않을 것 같네요.

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/3c5e369f9659e5635360965a93ce2447ae442734

수정완료했습니다 ☺

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+
+    @PutMapping("/reset")
+    @ResponseStatus(HttpStatus.RESET_CONTENT)
+    private void resetData(@RequestBody Map<String, Long> request) {
```
Map으로 요청을 받지는 말아주세요.
어떤 데이터가 들어올지 코드를 읽는 사람은 모릅니다.

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/3c5e369f9659e5635360965a93ce2447ae442734

해당 리퀘스트 전용 DTO를 제작하는게 옳은지 모르겠지만, 위와 같이 개선하였습니다!

ksundong replied:
옳은지 않은지 한 번 생각해주세요.

ksundong commented on be/src/main/java/com/team22/baseball/controller/ApiGameController.java:
```diff
@@ -0,0 +1,75 @@
+    @PutMapping("/reset")
+    @ResponseStatus(HttpStatus.RESET_CONTENT)
+    private void resetData(@RequestBody Map<String, Long> request) {
+       gameService.resetGameData(request.get("gameId"));
```
gameId 를 path variable로 받고
`POST /api/games/{gamdId}/reset`
이 더 깔끔하지 않을까요?

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/0a830a0f3ccb8a0b1af5a20099c1dc0c78d1eb26

왜 저는 개선하다고 쓸데없이 DTO를 만들어서 RequestBody로 가져왔을까요..

불필요한 DTO도 안만들 수 있고, 가독성이 좋은 URL 설계 정말 중요한 것 같습니다..

ksundong commented on be/src/main/java/com/team22/baseball/domain/Game.java:
```diff
@@ -0,0 +1,55 @@
+public class Game {
+
+    @Id
+    private final Long id;
+
+    private final int round;
+
+    private final boolean inProgress;
```
불변하게 상태 만들어주신거 좋네요.

ksundong commented on be/src/main/java/com/team22/baseball/domain/Game.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @MappedCollection(idColumn = "game_id", keyColumn = "id")
+    private List<Team> teams = new ArrayList<>();
```
필드 - 생성자 - 메서드 순으로 정렬시켜주는게 자바 관례입니다.

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/7021ee8e43bc81d7e0ce962fb4cca5e2e5af09b8

다시한번 리마인드 시켜주셔서 감사합니다.
반영 완료했습니다!

ksundong commented on be/src/main/java/com/team22/baseball/domain/Game.java:
```diff
@@ -0,0 +1,55 @@
+        return teams;
+    }
+
+    public void addTeam(Team team) {
+        teams.add(team);
+    }
```
되도록 getter는 최하단에 모아주시고, 로직을 전개하기 위해 필요한 메서드를 위로 모아주세요.

ksundong commented on be/src/main/java/com/team22/baseball/domain/Game.java:
```diff
@@ -0,0 +1,55 @@
+
+    public static Game of(int round, boolean inProgress) {
+        return new Game(null, round, inProgress);
+    }
```
static 메서드는 메서드 맨 위에 두는 것이 관례입니다.
그리고 null을 인자로 받기보다는, 생성자를 하나 더 만들어주세요.

ksundong commented on be/src/main/java/com/team22/baseball/domain/Player.java:
```diff
@@ -0,0 +1,102 @@
+@Table(value = "PLAYER")
+public class Player {
+
+    // TODO. 추후 PlayerScore 클래스를 만들어서 멤버변수 분리 예정
```
좋은 TODO네요.

ksundong commented on be/src/main/java/com/team22/baseball/domain/PlayerHistory.java:
```diff
@@ -0,0 +1,61 @@
+
+    public PlayerHistory of(Long playerId, int round, String record) {
+        return new PlayerHistory(null, playerId, round, record);
+    }
```
많이 바쁘셨군요. ㅋㅋ

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/3b3d6a561a99864ae0321b3bb024d17d370f0d99

앗...너무 창피한 실수네요..제거완료했습니다.

ksundong commented on be/src/main/java/com/team22/baseball/domain/TeamScore.java:
```diff
@@ -0,0 +1,55 @@
+public class TeamScore {
+
+    @Id
+    @JsonIgnore
+    private final Long id;
+
+    @JsonIgnore
+    private final Long teamId;
```
도메인 객체에 Jackson 관련 애노테이션이 붙는건 이상한 상황이겠죠?

sanhee replied:
DTO 만들기전에 도메인 리턴했을 때 붙여뒀던 부분인데 미처 지우지 못했습니다.

지적해주셔서 감사합니다!

ksundong commented on be/src/main/java/com/team22/baseball/dto/request/UpdatePlayerInfo.java:
```diff
@@ -0,0 +1,67 @@
+    private boolean hit;
+
+    @JsonCreator
+    public UpdatePlayerInfo(@JsonProperty("round") int round, @JsonProperty("teamName") String teamName, @JsonProperty("teamScore") int teamScore, @JsonProperty("playerName") String playerName, @JsonProperty("hit") boolean hit) {
```
`JsonCreator`는 굳이 필요없지 않을까요?

sanhee replied:
없어도 잘 동작하네요!

JSON 인식이 잘안됐을때  스택오버플로우 샘플을 참고한건대 무작점 복붙하는 습관은 줄여야겠습니다.

ksundong commented on be/src/main/java/com/team22/baseball/dto/response/DetailScore/DetailScore.java:
```diff
@@ -0,0 +1,45 @@
+        return name;
+    }
+
+    public int[] getScores() {
+
+        //INFO. SCORES 는 라운드 정보를 포함하고 있는 동적 객체여서 그대로 보낼 수도 있지만, 프론트단에서 동적으로 파싱하기 불편하여 12 사이즈의 배열로 변환함
+
+        int[] arrayScore = new int[10];
+        Arrays.fill(arrayScore, -1); // 점수가 없는건 -1
+
+        for (int i = 0; i < scores.size(); i++) {
+            arrayScore[i] = scores.get(i).getScore();
+        }
+
+        return arrayScore;
+    }
```
이 부분은 좀 더 생각해보세요~

ksundong replied:
List를 그대로 반환하지말고, 새로운 List를 만들어서 반환해주면 되겠죠?
굳이 배열을 쓸 이유는 없어보여요.

sanhee replied:
뭔가 배열이 아직도 익숙해서...습관적으로 작성했었는데,, 고쳐보도록 하겠습니다!

ksundong commented on be/src/main/java/com/team22/baseball/dto/response/DetailScore/DetailScore.java:
```diff
@@ -0,0 +1,45 @@
+
+    public DetailScore(String name, List<TeamScore> scores) {
+        this.name = name;
+        this.scores = scores;
```
놓치기 쉬운 문제가 있는데 scores가 null이면 null이게 됩니다.
저도 취업하고나서 알게되었네요.

sanhee replied:
메소드의 인자로 null값이 들어오면 null로 초기화가 된다는 말씀이신가요???
따로 유효성 검사 로직을 해야하는 건지 다른 좋은 방법이 있는지 궁금합니다!

ksundong replied:
null이면 빈 컬렉션이 할당되도록 해주는게 지금까진 제일 나은 것 같습니다.

ksundong commented on be/src/main/java/com/team22/baseball/dto/response/GameList/GameList.java:
```diff
@@ -0,0 +1,58 @@
+    private String home;
+
+    @JsonProperty("away")
+    private String away;
```
요건 굳이 해주실 필요 없지 않을까요?

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/818b853270d3dc2cc871a89f8a4acc35ebef6f25
출력되는 JSON 순서가 맞지 않아서, 억지로 써줬는데 JSON은 순서와 관련된 데이터가 아니니 의미가 없는 프로퍼티였네요. 제거하도록 하겠습니다.

ksundong commented on be/src/main/java/com/team22/baseball/dto/response/PlayerScoreList/PlayerInfo.java:
```diff
@@ -0,0 +1,73 @@
+    private int hits;
+
+    @JsonProperty("outs")
+    private int outs;
```
모두가 같은 스네이크 케이스라면
Spring의 Jackson 설정을 바꿔주세요.

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/d756fad447b33372425a5c91fd632a3541a5c0bf

오 이런 꿀팁이 있었군요!!

제 손목건강이 +1 향상될 것 같습니다.👍

ksundong commented on be/src/main/java/com/team22/baseball/dto/response/TeamSelect/PlayerInfoDto.java:
```diff
@@ -0,0 +1,55 @@
+    @JsonProperty("uniform_number")
+    private int uniformNumber;
+
+    @JsonIgnore   // 왜인지는 모르겠지만 boolean 형식의 자료형은 Json필드가 하다 더 생김... 이건 pitcher이름으로 생김 그래서 무시옵션
+    private boolean isPitcher;
```
왜일까요~?

ksundong replied:
저희 회사에서 JsonProperty 지정해서 해결해버리시더라고요 ㅋㅋ

sanhee replied:
https://shuaibabdulla40.medium.com/jackson-duplicate-boolean-fields-in-json-525c67588407 
이런 문제가 있군요..다음 프로젝트때 적용해보도록 하겠습니다.

ksundong commented on be/src/main/java/com/team22/baseball/dto/response/TeamSelect/TeamListDto.java:
```diff
@@ -0,0 +1,43 @@
+    private TeamInfoDto teamInfoDto;
+
+    @JsonProperty("player")
+    private List<PlayerInfoDto> playerInfoDtoList;
```
보통 변수명에는 자료형을 붙여주지 않는게 좋습니다.

ksundong commented on be/src/main/java/com/team22/baseball/repository/GameRepository.java:
```diff
@@ -0,0 +1,98 @@
+
+    List<Game> findAll();
+
+    @Query("SELECT * FROM GAME WHERE GAME.id = :id;")
```
스타의 사용은 무조건 지양해주세요.
왜 안되는 지는 호눅스에게 질문해주세요 ㅋㅋ

sanhee replied:
DB 수업때 들은 기억이 있는데, 희미해서 잊혀질뻔했네요.
지적해주셔서 감사합니다!!

ksundong commented on be/src/main/java/com/team22/baseball/repository/GameRepository.java:
```diff
@@ -0,0 +1,98 @@
+
+public interface GameRepository extends CrudRepository<Game, Long> {
+
+    Optional<Game> findById(Long id);
+
+    List<Game> findAll();
```
Override 애노테이션 붙여주세요.

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/a4e0a44838332c206d065716d527d936c594768a
깜박했었네요!! 수정했습니다. ㅎㅎ

ksundong commented on be/src/main/java/com/team22/baseball/repository/GameRepository.java:
```diff
@@ -0,0 +1,98 @@
+    @Query("SELECT * FROM GAME WHERE GAME.id = :id;")
+    Optional<Game> findGameById(@Param("id") Long id);
+
+    @Query("SELECT TEAM.id as id, TEAM.name as name, TEAM.is_home as is_home, TEAM.selected as selected, TEAM.game_id as game_id FROM GAME INNER JOIN TEAM ON GAME.id = TEAM.game_id AND TEAM.name = :title;")
+    Optional<Team> findTeamByTitle(@Param("title") String title);
+
+    @Query("SELECT * FROM PLAYER AS P WHERE P.name = :name;")
+    Optional<Player> findPlayerByName(@Param("name") String name);
```
고민인게, Repository 하나에서 접근한다는게 이상적인지 잘 모르겠어요.
Repository를 나누는 접근이 더 좋을 것 같다는 개인적인 의견입니다.

sanhee replied:
저도 코드작성을 하면서 이게 진정 올바른건가? 라는 생각이 끊임없었습니다..
저도 역할에 따라 나누는 방향을 생각해보도록 하겠습니다! 감사합니다!

ksundong commented on be/src/main/java/com/team22/baseball/repository/GameRepository.java:
```diff
@@ -0,0 +1,98 @@
+            "FROM GAME INNER JOIN TEAM ON GAME.id = TEAM.game_id\n" +
+            "WHERE TEAM.is_home = false) as away_group\n" +
+            "ON home_group.id = away_group.id;")
+    List<GameList> findAllGame();
```
너무 길죠?
당연히 문자열이기 때문에 컴파일 타임에 검증이 안될겁니다.
어쩌면 IntelliJ가 지원해주는 Query Support도 제대로 작동이 안될 수도 있겠네요.
거기에 `\n`까지 있으니, 더더욱이요.

저는 multiline string을 지원하는 kotlin을 사용했었고요.
이것을 쓸 때에는 JdbcTemplate을 이용했었습니다.

sanhee replied:
이번 프로젝트부터 JDBC템플릿을 사용하게 되는데, 그때 잘 학습해보도록 하겠습니다!!

ksundong commented on be/src/main/java/com/team22/baseball/repository/GameRepository.java:
```diff
@@ -0,0 +1,98 @@
+            "ON home_group.id = away_group.id;")
+    List<GameList> findAllGame();
+
+    @Query("SELECT TEAM.name FROM TEAM WHERE TEAM.game_id = (SELECT TEAM.game_id FROM TEAM WHERE TEAM.name = :title);")
```
JOIN을 주로 사용해주시고, 서브쿼리는 지양해주세요.

ksundong commented on be/src/main/java/com/team22/baseball/service/GameService.java:
```diff
@@ -0,0 +1,193 @@
+@Service
+public class GameService {
+
+    private final int ONE = 1;
```
static을 써주세요.

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/a4e0a44838332c206d065716d527d936c594768a
수정했습니다!!

ksundong commented on be/src/main/java/com/team22/baseball/service/GameService.java:
```diff
@@ -0,0 +1,193 @@
+
+        }
+
+        return teamListDtos;
```
왜 다 사회적 거리두기를 하고 있나요?

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/482336f637d172884fb6b154e37ffeb4fc942f9f

ㅋㅋㅋㅋㅋㅋㅋㅋㅋ시국이 시국인지라...코드 습관까지 이렇게 됐나봅니다 ㅋㅋㅋㅋㅋㅋ
수정완료했습니다.☺☺

ksundong commented on be/src/main/java/com/team22/baseball/service/GameService.java:
```diff
@@ -0,0 +1,193 @@
+
+            teamListDtos.add(new TeamListDto(teamInfoDto, playerInfoDto));
+
+        }
```
stream으로 개선해보세요.

ksundong commented on be/src/test/java/com/team22/baseball/service/GameServiceTest.java:
```diff
@@ -0,0 +1,100 @@
+
+@SpringBootTest
+@Transactional
+class GameServiceTest {
```
과연 Service Test일까요? ㅋㅋ

sanhee replied:
아직 테스트 개념이 없다보니.. 주먹구구식으로 작성했습니다.

다음 프로젝트때는 좀 더 학습을 하고 진행하도록 하겠습니다.

ksundong commented on be/src/test/java/com/team22/baseball/service/GameServiceTest.java:
```diff
@@ -0,0 +1,100 @@
+    @Test
+    @DisplayName("첫번째 게임을 검색한다.")
+    void findGameOfId() throws Exception {
+        Game findGame = gameRepository.findById(1L).orElseThrow(Exception::new);
+        assertThat(findGame).isNotNull();
```
외부의 상태에 의존적인 테스트는 좋지 않습니다.

ksundong commented on be/src/main/java/com/team22/baseball/service/GameService.java:
```diff
@@ -0,0 +1,193 @@
+        return gameRepository.findPlayerByName(name).orElseThrow(Exception::new);
+    }
+
+    public int[] calculatePlayerScore(UpdatePlayerInfo req) throws Exception {
+
+        Player findPlayer = findPlayerByName(req.getPlayerName());
+
+        int plateAppearance = findPlayer.getPlateAppearance() + ONE;
+        int hits;
+        int outs;
+
+        if (req.isHit()) {
+            hits = findPlayer.getHits() + ONE;
+            outs = findPlayer.getOuts();
+        } else {
+            outs = findPlayer.getOuts() + ONE;
+            hits = findPlayer.getHits();
+        }
+
+        return new int[]{plateAppearance, hits, outs};
+
+    }
```
배열 사용은 지양해주세요~

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/65ac392d91da705d455351b5c8f928baaeba70d4

리스트 형식으로 변경완료했습니다!

ksundong commented on be/src/main/java/com/team22/baseball/service/GameService.java:
```diff
@@ -0,0 +1,193 @@
+        int hits;
+        int outs;
+
+        if (req.isHit()) {
+            hits = findPlayer.getHits() + ONE;
+            outs = findPlayer.getOuts();
+        } else {
+            outs = findPlayer.getOuts() + ONE;
+            hits = findPlayer.getHits();
+        }
```
else문을 없애는 방향으로 로직을 진행해보세요.

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/65ac392d91da705d455351b5c8f928baaeba70d4

기본적으로 아웃상태로 변수를 초기화하고, 안타일 경우에만 수정하도록 하여 `else`문을 제거하였습니다.

지적해주셔서 감사합니다.

ksundong commented on be/src/main/java/com/team22/baseball/service/GameService.java:
```diff
@@ -0,0 +1,193 @@
+
+    }
+
+    public Game findGameById(Long gameId) throws Exception {
+        return gameRepository.findGameById(gameId).orElseThrow(Exception::new);
+    }
+
+    public NextPlayerInfoDto findNextPlayerByNumberAndTeamName(int nextUniformNumber, String teamName) throws Exception {
+        return gameRepository.findNextPlayerByNumberAndTeamName(nextUniformNumber, teamName).orElseThrow(Exception::new);
+    }
```
Exception 던지기는 하면안됩니다~

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/35e016331085b977a3810307fd7b5b35e7ed2335

개발당시에 절대 그러면 안되지만...빨리하자라는 생각에 Exception 던지기를 해버린 것 같습니다..

GlobalExceptionHandler를 생성하였습니다.

ksundong commented on be/src/main/java/com/team22/baseball/service/GameService.java:
```diff
@@ -0,0 +1,193 @@
+            TeamInfo teamInfo = TeamInfo.of(team.getName(), team.isHome(), team.isSelected());
+
+            for (Player player : team.getPlayers()) {
+                playerInfos.add(PlayerInfo.of(player.getUniformNumber(), player.getName(), player.getPlateAppearance(), player.getHits(), player.getOuts()));
```
이럴 땐 빌더패턴을 써봅시다.
아니면 `new PlayerInfo(player)`는 어떤가요?

sanhee replied:
https://github.com/codesquad-members-2021/baseball/pull/44/commits/b255078eb1df1866d6f0bd554a79938aae6ad04a

행으로 너무 길어져서 가독성이 심각하게 떨어졌는데, Player를 생성자 인자로 받으면 효과적으로 코드를 줄일 수 있군요!!!
이런 발상은 생각해본적이 없는데 조언해주셔서 너무 감사합니다!!

ksundong commented on be/src/main/java/com/team22/baseball/service/GameService.java:
```diff
@@ -0,0 +1,193 @@
+
+    }
+
+    public NextPlayerInfoDto updatePlayerInfo(UpdatePlayerInfo req) throws Exception {
+
+        int[] scores = calculatePlayerScore(req);
+
+        final int PLATE_APPEARANCE = 0;
+        final int HITS = 1;
+        final int OUTS = 2;
+
+        final int MAX_PLAYER = 10;
+
+        updatePlayerInfo(req.getPlayerName(), scores[PLATE_APPEARANCE], scores[HITS], scores[OUTS]);
+        insertTeamScore(req.getTeamName(), req.getRound(), req.getTeamScore());
+
+        Player prevPlayer = findPlayerByName(req.getPlayerName());
+
+        int prevUniformNumber = prevPlayer.getUniformNumber();
+        String teamName = req.getTeamName();
+
+        int nextUniformNumber = ++prevUniformNumber > MAX_PLAYER ? ONE : prevUniformNumber;
+
+        return findNextPlayerByNumberAndTeamName(nextUniformNumber, teamName);
+    }
```
이 로직 너무 길지 않나요?
상수가 막 지역변수로 등장하기도 하고, 좀 그렇죠?
개선해봅시다.

ksundong reviewed (changes requested):
안녕하세요! 노을!! 리뷰어 Dion입니다.
원래 야구게임 자체가 제가 코드스쿼드에서 해왔던 프로젝트중에 난이도가 아마 가장 높지 않나 싶었던 프로젝트였어요.
Spring Data JDBC만을 사용해야한다는 제약도 있고, 도메인 구현자체도 조금 까다로웠습니다.

따라서 이번 미션은 구현만해도 어느정도 성공이라고 봅니다.
다만, 코드 퀄리티에 있어서는 피드백을 드리려고 합니다.

읽는 사람을 배려하는 코드를 작성해보면 어떨까요? 시간이 없고, 촉박할 때 중요한건 습관입니다.
습관적으로 클린코드를 작성하도록 해야 바쁠때에도 클린코드를 작성할 수 있다고 생각해요.

API 협의에 대해서 적혀있어서 개인적인 생각을 말씀드릴게요.
API를 협의하기 이전에, '우리 애플리케이션이 어떤식으로 동작할 것인가'에 대한 상호 협의는 필요합니다.
그래야 뭘 주고 뭘 받을지, 어떤걸 어디서 처리할 지 결정할 수 있거든요.
다만, API를 협의하는건 API를 설계한 이후가 좋은 것 같아요. 내가 만든 API 설계가 별로인 부분만 피드백을 주면, 개선하면 되거든요.
그리고 보통 좋은 설계는 바로 나오지 않기 때문에, API를 협의하다보면 개발을 할 수 있는 시간마저 잃게됩니다.
적당히 스타트를 끊고, 지속적으로 개선해나가는 것이 보다 애자일하고, 좋은 설계가 나올 수 있는 바탕이 됩니다.

> 질문 드릴점은 아래와 같이 Game Repostiory에서 반환값으로 Game외에 다른 객체들과 DTO들을 전달하고 있는데 괜찮은 방식인지 궁금합니다.

JPA를 사용할 때도, DTO Projection이라고 DTO로 바로 리턴하는 기능을 사용하곤 합니다.
도메인에 대한 조작이 필요없고 바로 조회만 한다면 바로 DTO로 리턴해줘도 괜찮다고 생각해요. 이건 어디까지나 제 생각이지만요.
가급적이면, 조회를 담당하는 Repository가 따로 있어도 괜찮겠죠.

> 또 컨트롤러의 response 반환값으로 주로 List<DTO> 형태를 사용하고 있는데 이 부분을 Wrapping 하라는 의미를 잘모르겠어서 약간의 도움을 주셨으면 좋겠습니다. 😁

좋은 질문이네요.
노을이 작성하신 List를 리턴하는 API의 응답은 다음과 같습니다.

```json
[
  {},
  {}
]
```

이런 형태의 경우는

1. 읽는 사람이 보기 불편합니다. url로 설명을 한다지만, 결국 body는 그냥 객체의 나열이거든요.
2. 만약 추가적인 정보가 필요한 경우 breaking change가 발생합니다.

그래서 제안드리는 형태는 다음과 같습니다.

```json
"anyList": [
  {},
  {}
]
```

만약 anyInfo라는것이 추가적으로 필요해진다면 다음과 같이 필드만 추가해주면 됩니다.

```json
"anyList": [
  {},
  {}
],
"anyInfo": {}
```

이런 형태로 만들기 위해서는 객체로 한 번 감싸주어야 합니다.

도움이 되셨나요?

---

전반적으로 코드퀄리티가 아쉬웠습니다. 그리고, 테스트 코드는 단위테스트를 작성하는 것부터 시도해보시기 바랍니다.
그리고 가급적이면 리뷰를 매 주 보내셔서 피드백을 받는 형태를 추천드릴게요.

아래의 코멘트를 참고하셔서 다음 프로젝트를 좀 더 잘 구현해보시면 좋을 것 같습니다. 2주간 고생많으셨습니다.
궁금하시거나, 도움이 필요한 부분, 이해가 되지 않는 부분등은 코멘트 남겨주세요~

sanhee committed:
style: 언더스코어로 작성된 URL 하이폰으로 변경

https://github.com/codesquad-members-2021/baseball/pull/44#discussion_r632926593

sanhee committed:
feat: Game Response 생성

- 기존 GameList는 의미가 모호해 GameInfo로 이름을 변경하였습니다.

sanhee committed:
refactor: findAllGame 중복 호출 제거

sanhee committed:
fix: selectGame 메소드 put 맵핑으로 변경

게임의 상태와 팀의 상태를 변경하는 컨트롤러로 기존 GET 보다는 PUT이 어울려 수정했습니다.

sanhee committed:
style: PUT 맵핑 메서드의 불필요한 동사 제거

sanhee committed:
refactor: Response Wrapping 객체 정의

sanhee committed:
refactor: calculatePlayerScore()의 else문 제거 및 리턴형식 배열 -> 리스트로 변환

sanhee committed:
refactor: getPlayerScoreOfGame() 파라미터로 gameId를 받도록 변경

sanhee committed:
style: Long gameID 카멜케이스에 맞게 명명

sanhee committed:
refactor: resetData() Post 맵핑으로 변경 및 Map 제거

추후 간편한 유효성 검사를 위해 클라이언트 Json 리퀘스트를 GameId DTO로 받도록 변경했습니다.

sanhee committed:
refactor: PlayerHistory 불필요한 메소드 제거

sanhee committed:
refactor: GameInfo의 불필요한 JsonProperty 제거

sanhee committed:
feat: Jackson 프로퍼티 네이밍 전략 : SNAKE CASE로 변경

sanhee committed:
style: 누락된 @Override, static 추가

sanhee committed:
style: 사회적 거리두기 해제

sanhee committed:
refactor: PlayerInfo를 Player를 통해 생성할 수 있게 정의

sanhee committed:
refactor: resetData() RequestBody -> PathVariable를 통해 gameId 로드

sanhee committed:
style: Game 클래스 [필드 - 생성자 - 메서드] 순 정렬

ksundong reviewed (approved):
고생하셨습니다!
다음 프로젝트도 파이팅해주세요!
